Sources/Build/misc.swift

```diff
@@ -43,6 +43,42 @@ extension CModule {
     }
 }
 
+extension String {
+    var isCpp: Bool {
+        guard let ext = self.fileExt else {
+            return false
+        }
+        return ext == "cpp"
+    }
+}
```

Comment:
This should reuse the extensions defined elsewhere -- we should break that into C++ and non-C++ groups so we can just reuse the same set here.